### ISSUE TITLE
`adding-field-with-default` (bug): `ARRAY[]` and `CURRENT_TIMESTAMP` are non-volatile

### ISSUE THREAD

**Issue by reteps** [sbdchd/squawk#645]
I am under the impression that `ARRAY[]` is non-volatile, which is causing false positives in our codebase.

Additionally, `CURRENT_TIMESTAMP` and related are stable according to https://www.postgresql.org/docs/current/xfunc-volatility.html

> Another important example is that the current_timestamp family of functions qualify as STABLE, since their values do not change within a transaction.

Config:

```toml
pg_version = "16.0"
assume_in_transaction = true
excluded_rules = []
```

Sample flagged migrations:

```sql
ALTER TABLE assessments
ADD COLUMN duration_stat_thresholds interval[] NOT NULL DEFAULT ARRAY[]::interval[];

ALTER TABLE assessments
ADD COLUMN duration_stat_threshold_labels text[] NOT NULL DEFAULT ARRAY[]::text[];

ADD COLUMN duration_stat_threshold_seconds double precision[] NOT NULL DEFAULT ARRAY[]::double precision[];

ADD COLUMN heartbeat_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP;

-- This might be hard to statically determine that it's non-volatile
ADD COLUMN statistics_last_updated_at timestamptz NOT NULL DEFAULT now() - interval '100 years';
```

**Comment by sbdchd** [comment]
Ah good find, we're checking at

https://github.com/sbdchd/squawk/blob/52fdeca5d5466957cc21be33bf399fc7bd763949/crates/squawk_linter/src/rules/adding_field_with_default.rs#L72-L73




we need to update either following to handle `current_timestamp` and `array[]`

https://github.com/sbdchd/squawk/blob/52fdeca5d5466957cc21be33bf399fc7bd763949/crates/squawk_linter/src/rules/adding_field_with_default.rs#L30-L51

or

https://github.com/sbdchd/squawk/blob/52fdeca5d5466957cc21be33bf399fc7bd763949/crates/squawk_linter/src/rules/adding_field_with_default.rs#L11-L17



I think we could probably handle the `now() - interval '100 years'` style case

**Comment by sbdchd** [comment]
Double checked that `array[]` is safe and it is:


```sql
CREATE TABLE bench_table (
    id SERIAL PRIMARY KEY,
    name VARCHAR(50),
    value INTEGER
);


INSERT INTO bench_table (name, value)
SELECT 
    'user_' || generate_series,
    (random() * 1000)::INTEGER
FROM generate_series(1, 10000000);



-- takes 6ms
ALTER TABLE bench_table 
ADD COLUMN empty_doubles double precision[] DEFAULT ARRAY[]::double precision[];


-- takes 28s
ALTER TABLE bench_table 
ADD COLUMN random_score NUMERIC DEFAULT random();
```

**Comment by sbdchd** [comment]
Thanks for the report! This is fixed in the latest version!

https://github.com/sbdchd/squawk/releases/tag/v2.26.0
